viewing_party/party.py

```diff
@@ -1,23 +1,205 @@
+        for movie_data in element["watched"]:
+                list_friends.append(movie_data)
+    
+    
+    for element in amandas_data["watched"]:
+            list_amandas.append(element)
+  
+
+    for element in list_friends:
+        if element not in list_amandas:
+            if element not in friends_unique_movies:
+                friends_unique_movies.append(element)
+    
+    return friends_unique_movies
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(amandas_data):
+
+    rec_list = []
+    friends_unique_movies =[]
+    list_friends = []
+    list_amandas = []
+
+    for element in amandas_data["friends"]:
+        for movie_data in element["watched"]:
+                list_friends.append(movie_data)
+    
+    
+    for element in amandas_data["watched"]:
+            list_amandas.append(element)
+
+
+    for element in list_friends:
+        if element not in list_amandas:
+            friends_unique_movies.append(element)
+
+    for movie in friends_unique_movies:
+        if movie["host"] in amandas_data ["subscriptions"]:
+            rec_list.append(movie)
+    
+    return rec_list
+    
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(sonyas_data):
+    rec_list_genre = []
+    friends_unique_movies =[]
+    list_friends = []
+    list_sonyas = []
+    fav_genre = []
+
+    for element in sonyas_data["friends"]:
+        for movie_data in element["watched"]:
+            list_friends.append(movie_data)
+    
+    if len (sonyas_data["watched"])== 0:
+        return rec_list_genre
+    for element in sonyas_data["watched"]:
+        list_sonyas.append(element)
+
+    if len(list_friends) == 0:
+        return rec_list_genre
+    for element in list_friends:
+        if element not in list_sonyas:
+            friends_unique_movies.append(element)
+
+    for movie in sonyas_data["favorites"]:
+        fav_genre.append(movie["genre"])
+
+    
+    for movie in friends_unique_movies:
+        if movie["host"] in sonyas_data ["subscriptions"]:
+            if movie["genre"] in fav_genre:
+                rec_list_genre.append(movie)
+
+    return rec_list_genre
+    
+def get_rec_from_favorites(sonyas_data):
+
+    rec_list_fav = []
+    sonyas_unique_movies =[]
+    list_friends = []
+    list_sonyas = []
+
+    for element in sonyas_data["friends"]:
+        for movie_data in element["watched"]:
+            list_friends.append(movie_data)
+    
+    
+    for element in sonyas_data["watched"]:
+        list_sonyas.append(element)
+
+    
+    for element in list_sonyas:
+        if element not in list_friends:
+            sonyas_unique_movies.append(element)
+
+    for movie in sonyas_unique_movies:
+        if movie not in list_friends:
+            rec_list_fav.append(movie)

```

Comment:
These two loops are doing the same work.  Both are eliminating movies that are in `list_friends`.